elasticgraph-schema_definition/lib/elastic_graph/schema_definition/schema_elements/scalar_type.rb

```diff
@@ -85,6 +86,10 @@ def initialize(schema_def_state, name)
           if missing.any?
             raise Errors::SchemaError, "Scalar types require `mapping` and `json_schema` to be configured, but `#{name}` lacks #{missing.join(" and ")}."
           end
+
+          inferred_grouping_missing_value_placeholder&.tap do |placeholder|
+            self.runtime_metadata = runtime_metadata.with(grouping_missing_value_placeholder: placeholder)
+          end
```

Comment:
```suggestion
          if (placeholder = inferred_grouping_missing_value_placeholder)
            self.runtime_metadata = runtime_metadata.with(grouping_missing_value_placeholder: placeholder)
          end
```

This approach is a bit more idiomatic (and is common throughout the codebase). `&.tap { ... }` works but can be a bit less efficient as it has to jump to the `tap` method, then jump back to the block you passed.

Also, the primary use for `tap` is to preserve a particular return value while being able to apply some kind of side effect on it, but `initialize` doesn't have a meaningful return value (it's treated a `void`).